File: src/BaseHmiTypes/Screens/Shapes/HmiUnkown.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiUnkown : HmiSurfaceShapeBase
{
    public HmiUnkown()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiUnkown;
    }

    public string? Type { get; set; }

    public string? SourceFormat { get; set; }

    public byte[]? SourceData { get; set; }

    public Dictionary<string, string> SourceProperties { get; } = [];
}

File: typescript/base-hmi-types/src/screens/shapes/HmiUnkown.ts
import { HmiSurfaceShapeBase } from "./HmiSurfaceShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiUnkown extends HmiSurfaceShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiUnkown;
  }

  type?: string;
  sourceFormat?: string;
  sourceData?: Uint8Array;
  readonly sourceProperties: Record<string, string> = {};
}
